csrc/scheduler/mma_utils.cpp

```diff
@@ -1810,13 +1810,14 @@ DimRolesMap MatmulPattern::getDimRoles(IdModel& id_model) const {
 
   } else if (output->definition()->isA<LinearOp>()) {
     const std::vector<IterDomain*>& out_logical = output->getLogicalDomain();
+    bool k_bcast = A->getLogicalDomain().back()->isBroadcast();
```

Comment:
This is good for now, but in the future I think we might want to generalize this pattern matching so that this doesn't even look like a matmul pattern at this point. I am thinking we could have a general stage where we find patterns, possibly in the unsegmented fusion, and those can be of various flavors, like pointwise, matmul, etc. Each of those pattern matchers can then do its own replacements inside the `schedule*` functions (or before). That way we could have a fusion that had a broadcast K like this _inside_ a fusion that has a _real_ matmul and still be able to schedule it as a single-matmul fusion; the pointwise pattern would translate to a broadcasted mul and the matmul pattern would be translated to an `MmaOp`. Anyway, as I said this is probably fine for now but I'm somewhat sure we'll currently have trouble generating a kernel for broadcast K so I'll probably need to check for this condition in the matmul scheduler's `canScheduleCompileTime`.

Reply:
Do you want to open an issue to keep track of this case for later?

Reply:
#2748